internal/csi-addons/rbd/replication.go

```diff
@@ -937,3 +945,37 @@ func checkVolumeResyncStatus(ctx context.Context, localStatus types.SiteStatus)
 
 	return nil
 }
+
+func getCurrentReplicationStatus(ctx context.Context,
+	mirrorGlobalStatus types.GlobalStatus,
+) (replication.GetVolumeReplicationInfoResponse_Status, string, error) {
+	// get the state of replication
+	localStatus, err := mirrorGlobalStatus.GetLocalSiteStatus()
+	if err != nil {
+		log.ErrorLog(ctx, err.Error())
+
+		return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "", err
+	}
+
+	localState := localStatus.GetState()
+
+	// check if mirroring state is up/down
+	isUp := localStatus.IsUP()
+
+	if isUp {
```

Comment:
how about flipping this to `if !isUp`? Return early on error conditions makes the code a little cleaner. It would cause the large `switch` to not be indented anymore.

Reply:
Makes sense, updated the condition now.